### ISSUE TITLE
OS.keyboard_get_scancode_from_physical() always uses the system's "first" keyboard, not the currently selected one

### ISSUE THREAD

**Issue by JustTemmie** [godotengine/godot#78144]
### Godot version

3.5.2.stable

### System information

Garuda Linux 6.3.6-zen1 - Gnome 44.2 - x86 Ryzen 7 5800X

### Issue description

`OS.keyboard_get_scancode_from_physical()` always uses keyboard index 0 in the OS, rather than using the currently selected keyboard

![image](https://github.com/godotengine/godot/assets/47639983/7a5007eb-3a64-447d-83c2-c52c4f9225f5)
here for example i have colemak selected, but godot will translate the scancodes to the dvorak layout.

i assume this is a linux-specific issue, though it might be an issue for all platforms

### Steps to reproduce

example 1 - expected behaviour: 
i have cleared all inputs in the project settings, only adding in 2 test actions that use the non-physical keys "e" and "n"
![image](https://github.com/godotengine/godot/assets/47639983/438005f7-e26e-42f7-9282-e48cfbd14a43)

and my system only has a single keyboard layout active - the US colemak layout
![image](https://github.com/godotengine/godot/assets/47639983/65015a49-6e7a-4503-9d0b-a063771b6d1c)

running this code below
```python
func _ready() -> void:
	# loop over every input action in the project
	for action in InputMap.get_actions():
		# loop over every key in said input action
		for i in (InputMap.get_action_list(action)):
			# print the digital key set in the editor, along with it's scan code
			print("qwerty keys: ", i.as_text(), " - ", i.scancode)
			# translate the scancode over to the current keyboard layout, from qwerty
			i.scancode = OS.keyboard_get_scancode_from_physical(i.scancode)
			# print the same info again
			print("colemak keys: ", i.as_text(), " - ", i.scancode)
			# seperator
			print("-----")
```
will print out the key, and the scancode of both keys, in both layouts
```
qwerty key: E - 69
colemak key: F - 70
-----
qwerty key: N - 78
colemak key: K - 75
```
this is the expected outcome. As the second keys are the colemak counterparts to the qwerty keys.

example 2:
with the keyboard layouts "US qwerty" and "US colemak" on the system, with colemak being the selected one
![image](https://github.com/godotengine/godot/assets/47639983/d62542d7-1c12-4b3f-bd22-d998a2833498)

the same code will print out
```
qwerty key: E - 69
colemak key: E - 69
-----
qwerty key: N - 78
colemak key: N - 78
```
meaning godot doesn't translate the keys at all, even though the colemak layout is the selected one, not the qwerty one

example 3:
adding the dvorak layout first, before both the other layouts
![image](https://github.com/godotengine/godot/assets/47639983/7a5007eb-3a64-447d-83c2-c52c4f9225f5)

prints out
```
qwerty key: E - 69
colemak key: Period - 46
-----
qwerty key: N - 78
colemak key: B - 66
```
now godot is correctly translating the keys into dvorak, only issue is - *i still have colemak selected*

to use the minimal reproduction project just download it, and run the project within godot whilst having multiple keyboard layouts added to your system.

### Minimal reproduction project

[Minimal reproduction project.zip](https://github.com/godotengine/godot/files/11725680/Minimal.reproduction.project.zip)

**Comment by bruvzg** [comment]
It's fixed by https://github.com/godotengine/godot/pull/75461 in 4.x, but fix was not cherry-picked for 3.x.

**Comment by AThousandShips** [comment]
Can't be directly picked afik, given changes to the Linux target

**Comment by JustTemmie** [comment]
ah, i see. oh well thanks for the info anyways!

**Comment by bruvzg** [comment]
It can't be cherry-picked directly, this function was moved to `DisplayServer`, I'll make check if everything is working as it should and make a PR tomorrow.

**Comment by akien-mga** [comment]
Fixed by #78169.
